src/Operations.jl

```diff
@@ -1112,32 +1121,64 @@ function gen_test_code(testfile::String; coverage=false)
 end
 
 function with_temp_env(fn::Function, temp_env::String)
-    cache = Base.active_project()
+    old_load_path = copy(LOAD_PATH)
+    old_active_project = Base.ACTIVE_PROJECT[]
     try
-        Pkg.API.activate(temp_env)
-        fn()
+        withenv("JULIA_LOAD_PATH" => temp_env) do
+            empty!(LOAD_PATH)
+            push!(LOAD_PATH, "@", "@v#.#", "@stdlib")
+            Base.ACTIVE_PROJECT[] = temp_env
```

Comment:
Wouldn't it be better if this ran in the spawned process?

Reply:
In the process we spawn for `test`? In what way would it be better? Are you referring to the whole `with_temp_env` function?